src/components/flowchart/flowchart.js

```diff
@@ -520,23 +529,21 @@ export class FlowChart extends Component {
       toSelectedNode,
     } = this.props;
 
+    // Handle metadata panel display or node click toggle
     displayMetadataPanel ? onLoadNodeData(id) : onToggleNodeClicked(id);
     toSelectedNode(node);
 
     const { from, to, range } = this.state.slicedPipelineState;
     // if both "from" and "to" are defined
     // then on a single node click, it should reset the sliced pipeline state
-    if (from !== null && to !== null) {
-      this.updateSlicedPipelineState(null, null, []);
+    if (from !== null && to !== null && !this.props.isSlicingPipelineApplied) {
```

Comment:
This condition is looks confusing as negative condition is checked first, We can keep the positive check first.